packages/opentelemetry-scope-base/src/types.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface BaseScopeManager {
+  /**
+   * Get the current active scope
+   */
+  active(): unknown;
+
+  /**
+   * Run the fn callback with object set as the current active scope
+   * @param scope Any object to set as the current active scope
+   * @param fn A callback to be imediately run within a specific scope
+   */
+  with<T>(scope: unknown, fn: T): void;
```

Comment:
Could you specify that the type for `fn` must be a function?

E.g.: `with(scope: unknown, fn: (...args: unknown[]) => unknown): void`?

Should `with` return what the executed function returns?

Reply:
I believe we could just use `with(scope: unknown, fn: Function): void` right ? I don't see the point of typing the args there